### PR TITLE
Check before deleting existing firewall rule

### PR DESCRIPTION
**Pull request by leovms**

### CODE REVIEW

File: Artifacts/windows-winrm/config-winrm.ps1 [hunk 1]
```diff
@@ -141,12 +141,19 @@ function Add-FirewallException
...
     )
 
-    # Delete an exisitng rule
-    netsh advfirewall firewall delete rule name="Windows Remote Management (HTTPS-In)" dir=in protocol=TCP localport=$Port | Out-Null
-    Handle-LastExitCode
+    $ruleName = "Windows Remote Management (HTTPS-In)"
+
+    # Determine if the rule already exists.
+    netsh advfirewall firewall show rule name=$ruleName | Out-Null
+    if ($LasExitCode -eq 0)
```
Comment (joerage) [inline Artifacts/windows-winrm/config-winrm.ps1]: typo: $LastExitCode

File: Artifacts/windows-winrm/config-winrm.ps1 [hunk 2]
```diff
@@ -141,12 +141,19 @@ function Add-FirewallException
...
+    {
+        # Delete the exisitng rule.
```
Comment (joerage) [inline Artifacts/windows-winrm/config-winrm.ps1]: typo: existing

File: Artifacts/windows-winrm/config-winrm.ps1 (hunk 1 above)
Comment (leovms) [reply Artifacts/windows-winrm/config-winrm.ps1]: Thank you, paste error. Fixing.

File: Artifacts/windows-winrm/config-winrm.ps1 (hunk 2 above)
Comment (leovms) [reply Artifacts/windows-winrm/config-winrm.ps1]: Done.

**Pull request merged**